components/proxy/plugins/corsorigin/cors_origin.go

```diff
@@ -54,9 +56,12 @@ func (m CorsOrigin) ServeHTTP(w http.ResponseWriter, r *http.Request, next caddy
 	var allowedOrigins []string
 	if m.AnyDomain {
 		allowedOrigins = []string{"*"}
-	} else {
+	} else if m.BaseDomain != "" {
 		allowedOrigins = []string{"*." + m.BaseDomain}
+	} else if len(m.AllowedOrigins) != 0 {
```

Comment:
Isn't this redudant to the `BaseDomain` behavior? :thinking:

Reply:
It's the inverse behaviour we want.

> and in this case because we're enabling the cors for the sub-domain (api.) but allowing traffic from the root domain, we cannot use the BaseDomain directive anyway.

Reply:
The current BaseDomain rule does the following:

Allow gitpod.io/api when Origin is `*.domain`

what we need is 
Allow api.gitpod.io when Origin is `domain` - here we cannot have the *. prefix

Reply:
Ah, we're just adding `*.baseDomain` [here](https://github.com/gitpod-io/gitpod/pull/13743/files#diff-dbbdcc7e3776c50401457b2e561d7f33336207e6c67c2f9726f7d49242b89677R60). :bulb: Adding `baseDomain` as well would do the trick as well.

But am fine with this change as well, as it allows for more precise config. :+1:

Reply:
Indeed, I also prefer very explicit when it comes to CORS rather than the implicit which BaseDomain does.